release/long_running_tests/workloads/serve_failure.py

```diff
@@ -104,10 +119,14 @@ def create_deployment(self):
         def handler(self, *args):
             return new_name
 
-        handler.deploy()
+        ray.get(self.random_killer.spare.remote(new_name))
+
+        handler.deploy(_blocking=True)
 
         self.deployments.append(new_name)
 
+        ray.get(self.random_killer.kill.remote(new_name))
+
     def verify_deployment(self):
         deployment = random.choice(self.deployments)
         for _ in range(100):
```

Comment:
@edoakes As we discussed, I thought about raising an error if this assertion fails. What happens if the deployment is down because the `RandomKiller` killed it? Would we still want this grace period?

Reply:
ideally we would give it a retry policy and fail if the request doesn't succeed after N retries. let's make that change separately to avoid conflating changes given this test is hard to debug